django/common/adapters/fhir_api.py

```diff
@@ -52,6 +63,30 @@ def create(self, resource_type: str, payload: dict, auth_token=None):
         response.raise_for_status()
         return response.json()
 
+    def validate(self, resource_type: str, payload: dict, auth_token=None):
+        """Calls the /<resource_type>/$validate endpoint of HAPI FHIR.
+        Note that this function does not raise an exception if the status is not 2XX.
+
+        Args:
+            resource_type (str): the resource type
+            payload (dict): the FHIR instance
+            auth_token ([type], optional): The authentication token to access FHIR API.
+            Defaults to None.
+
+        Returns:
+            (dict): OperationOutcome containing the details about validation errors.
+        """
+        headers = {**self._headers, "Authorization": f"Bearer {auth_token}"} if auth_token else self._headers
+        response = requests.post(
+            f"{self._url}/{resource_type}/$validate",
+            json=payload,
+            headers=headers,
+        )
+        try:
+            return response.json()
+        except Exception:
+            response.raise_for_status()
```

Comment:
When does it happen? And what happens here?

Reply:
when the response is not json-serializable (the status doesn't matter)
just like in create or retrieve, this raises an exception according to the HTTP error https://docs.python-requests.org/en/master/api/#requests.Response.raise_for_status

Reply:
why the try except ?

Reply:
because if the response is "500 internal server error", I don't want the error raised to be "object is not JSON-serializable" but the original error "500 internal server error". So I catch the JSON error and re-throw the error from the response. I don't find it pretty either.
Any ideas ?